Non-deterministic `AssertionError: !(any(isnan, component_data))` when running the `von_misses_truss` example with ONSAS v0.4.6 on Julia 1.11

I am able to reproduce this with ONSAS v0.4.6 on Julia 1.11.6.

## MWE

Open Julia 1.11.6, and run the following code:

```julia
import Pkg

Pkg.add(name = "ONSAS", version = v"0.4.6")
```

Close Julia, and then open a new Julia session, and run the following code:

```julia
import ONSAS

julia_cmd = Base.julia_cmd()
script = pkgdir(ONSAS, "examples", "von_misses_truss", "von_misses_truss.jl")
cmd = `"$julia_cmd" "$script"`

for i in 1:100
    @info "Starting iteration $i"
    run(cmd)
    @info "Finished iteration $i"
end
```

In my experience, I can make it up to e.g. 5-10 iterations, but eventually I always get the failure, which looks like this:

```
Test Summary:                                                    | Pass  Total  Time
Analytic and numeric load factors case: RotatedEngineeringStrain |    1      1  0.1s
ERROR: LoadError: AssertionError: !(any(isnan, component_data))
Stacktrace:
  [1] write_cell_data(vtk::VTKMeshFile{WriteVTK.DatasetFile}, celldata::Vector{Matrix{Float64}}, name::String; component_names::Vector{String}, kwargs::@Kwargs{})
    @ ONSAS.VTK ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:137
  [2] write_cell_data
    @ ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:128 [inlined]
  [3] (::ONSAS.VTK.var"#24#25"{Dict{Symbol, Vector{Matrix{Float64}}}, Dict{Symbol, Vector{Float64}}})(vtx::VTKMeshFile{WriteVTK.DatasetFile})
    @ ONSAS.VTK ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:232
  [4] VTKMeshFile(::ONSAS.VTK.var"#24#25"{Dict{Symbol, Vector{Matrix{Float64}}}, Dict{Symbol, Vector{Float64}}}, ::String, ::Vararg{Any}; kwargs::@Kwargs{append::Bool})
    @ ONSAS.VTK ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:40
  [5] VTKMeshFile
    @ ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:37 [inlined]
  [6] #write_vtk#23
    @ ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:225 [inlined]
  [7] write_vtk
    @ ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:215 [inlined]
  [8] write_vtk(sol::Solution{StaticState, NonLinearStaticAnalysis{Structure{3, Mesh{3, Node{3, Float64}, Truss{3, GreenStrain, Float64, Node{3, Float64}, StaticArraysCore.SVector{2, Node{3, Float64}}}, AbstractFace, Nothing}, AbstractMaterial, Truss{3, GreenStrain, Float64, Node{3, Float64}, StaticArraysCore.SVector{2, Node{3, Float64}}}, AbstractBoundaryCondition, AbstractBoundaryCondition}, Float64, Vector{Float64}}, NewtonRaphson}, base_filename::String; fields::Vector{Symbol}, append::Bool)
    @ ONSAS.VTK ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:260
  [9] write_vtk(sol::Solution{StaticState, NonLinearStaticAnalysis{Structure{3, Mesh{3, Node{3, Float64}, Truss{3, GreenStrain, Float64, Node{3, Float64}, StaticArraysCore.SVector{2, Node{3, Float64}}}, AbstractFace, Nothing}, AbstractMaterial, Truss{3, GreenStrain, Float64, Node{3, Float64}, StaticArraysCore.SVector{2, Node{3, Float64}}}, AbstractBoundaryCondition, AbstractBoundaryCondition}, Float64, Vector{Float64}}, NewtonRaphson}, base_filename::String)
    @ ONSAS.VTK ~/.julia/packages/ONSAS/49upX/src/Interfaces/VTK.jl:251
 [10] run()
    @ Main ~/.julia/packages/ONSAS/49upX/examples/von_misses_truss/von_misses_truss.jl:135
 [11] top-level scope
    @ ~/.julia/packages/ONSAS/49upX/examples/von_misses_truss/von_misses_truss.jl:140
in expression starting at /Users/dilum/.julia/packages/ONSAS/49upX/examples/von_misses_truss/von_misses_truss.jl:140
ERROR: failed process: Process(`/path/to/julia/1.11.6/bin/julia -C native -J/path/to/julia/1.11.6/lib/julia/sys.dylib -g1 /Users/dilum/.julia/packages/ONSAS/49upX/examples/von_misses_truss/von_misses_truss.jl`, ProcessExited(1)) [1]

Stacktrace:
 [1] pipeline_error
   @ ./process.jl:598 [inlined]
 [2] run(::Cmd; wait::Bool)
   @ Base ./process.jl:513
 [3] run(::Cmd)
   @ Base ./process.jl:510
 [4] top-level scope
   @ ./REPL[5]:3
```

--- a/examples/cylinder_internal_pressure/cylinder_internal_pressure.jl
+++ b/examples/cylinder_internal_pressure/cylinder_internal_pressure.jl
@@ -10,7 +10,6 @@ Random.seed!(1234)
 # Mesh with Gmsh.jl
 include("cylinder_mesh.jl")
 
-"Return problem parameters"
 function parameters()
     ## scalar parameters (dimensions in mm an MPa)
     Lz = 30                 # cylinder length in z mm

--- a/examples/linear_extension/linear_extension.jl
+++ b/examples/linear_extension/linear_extension.jl
@@ -7,7 +7,6 @@ using ONSAS
 # Mesh with Gmsh.jl (see linear_extension_sketch)
 include("linear_extension_mesh.jl")
 
-"Return problem parameters"
 function parameters()
     E = 2.0                             # Young modulus in Pa
     ν = 0.4                             # Poisson's ratio

--- a/examples/self_weight_clamped_truss/self_weight_clamped_truss.jl
+++ b/examples/self_weight_clamped_truss/self_weight_clamped_truss.jl
@@ -29,7 +29,6 @@ abstract type AbstractCase end
 struct FirstCase <: AbstractCase end
 struct SecondCase <: AbstractCase end
 
-"Return problem parameters"
 function parameters()
     N = 4       # Number of elements.
     E = 30e6    # Young's modulus.

--- a/examples/uniaxial_compression/uniaxial_compression.jl
+++ b/examples/uniaxial_compression/uniaxial_compression.jl
@@ -7,7 +7,6 @@ using ONSAS
 # Mesh Cube with Gmsh.jl
 include(joinpath("..", "uniaxial_extension", "uniaxial_mesh.jl"))
 
-"Return problem parameters"
 function parameters()
     E = 1.0                     # Young modulus in Pa
     ν = 0.3                     # Poisson's ratio

--- a/src/Entities/Tetrahedrons.jl
+++ b/src/Entities/Tetrahedrons.jl
@@ -139,14 +139,14 @@ end
 
 function _B_mat!(B::Matrix, deriv::Matrix, F::Matrix)
     B[1:3,
-        :] = [diagm(deriv[:, 1]) * F' diagm(deriv[:, 2]) *
-                                      F' diagm(deriv[:, 3]) * F' diagm(deriv[:, 4]) * F']
+    :] = [diagm(deriv[:, 1]) * F' diagm(deriv[:, 2]) *
+                                         F' diagm(deriv[:, 3]) * F' diagm(deriv[:, 4]) * F']
 
     for k in 1:4
         B[4:6,
-            (k - 1) * 3 .+ (1:3)] = [deriv[2, k] * F[:, 3]' + deriv[3, k] * F[:, 2]'
-                                     deriv[1, k] * F[:, 3]' + deriv[3, k] * F[:, 1]'
-                                     deriv[1, k] * F[:, 2]' + deriv[2, k] * F[:, 1]']
+        (k - 1) * 3 .+ (1:3)] = [deriv[2, k] * F[:, 3]' + deriv[3, k] * F[:, 2]'
+                                        deriv[1, k] * F[:, 3]' + deriv[3, k] * F[:, 1]'
+                                        deriv[1, k] * F[:, 2]' + deriv[2, k] * F[:, 1]']
     end
     B
 end

--- a/src/Materials/NeoHookeanMaterial.jl
+++ b/src/Materials/NeoHookeanMaterial.jl
@@ -118,7 +118,7 @@ function _∂S∂E!(
     for index in INDEXES_TO_VOIGT
         i, j = index
         ∂S∂E[row,
-            :] .= voigt(
+        :] .= voigt(
             ForwardDiff.gradient!(aux_gradients,
                 E -> S_analytic(m, E)[i, j],
                 collect(𝔼)),

--- a/src/Meshes/Searches.jl
+++ b/src/Meshes/Searches.jl
@@ -85,26 +85,27 @@ contains the indexes of the elements containing the points."
 function evaluate_points_in_elements(elements::Vector{E}, vec_points::Vector{P},
         ::Threaded) where {dim, E <: AbstractElement{dim},
         P <: Point{dim}}
-    numthreads = Threads.nthreads()
-    in_mesh_points_idx = [Vector{Int64}() for _ in 1:numthreads]
-    in_mesh_elements_idx = [Vector{Int64}() for _ in 1:numthreads]
     num_points = length(vec_points)
+    # Store per-point results: 0 means not found, >0 is the element index.
+    result_elem_idx = zeros(Int64, num_points)
     Threads.@threads for point_idx in 1:num_points
-        id = Threads.threadid()
         point = vec_points[point_idx]
-        @inbounds for (elem_idx, elem) in enumerate(elements)
-            @inbounds if point ∈ elem
-                push!(in_mesh_points_idx[id], point_idx)
-                push!(in_mesh_elements_idx[id], elem_idx)
-                # Continue with next point, without checking the following elements.
+        for (elem_idx, elem) in enumerate(elements)
+            if point ∈ elem
+                result_elem_idx[point_idx] = elem_idx
                 break
             end
         end
     end
-    # Reorder in_mesh_elements_idx
-    in_mesh_points_idx = reduce(vcat, in_mesh_points_idx)
-    in_mesh_elements_idx = reduce(vcat, in_mesh_elements_idx)
-    # _sort_indexes!(in_mesh_points_idx, in_mesh_elements_idx)
+    # Collect results
+    in_mesh_points_idx = Vector{Int64}()
+    in_mesh_elements_idx = Vector{Int64}()
+    for point_idx in 1:num_points
+        if result_elem_idx[point_idx] != 0
+            push!(in_mesh_points_idx, point_idx)
+            push!(in_mesh_elements_idx, result_elem_idx[point_idx])
+        end
+    end
     in_mesh_points_idx, in_mesh_elements_idx
 end
 
@@ -196,11 +197,9 @@ function evaluate_points_in_mesh(mesh::AbstractMesh,
         end
     end
 
-    numthreads = Threads.nthreads()
-    in_mesh_points_idx = [Vector{Int64}() for _ in 1:numthreads]
-    in_mesh_elements_idx = [Vector{Int64}() for _ in 1:numthreads]
-    Threads.@threads for point_idx in 1:length(vec_points)
-        id = Threads.threadid()
+    num_points = length(vec_points)
+    result_elem_idx = zeros(Int64, num_points)
+    Threads.@threads for point_idx in 1:num_points
         point = vec_points[point_idx]
         @inbounds for (box_idx, Hi) in enumerate(Hpart)
             if point ∈ Hi
@@ -208,23 +207,25 @@ function evaluate_points_in_mesh(mesh::AbstractMesh,
                 found = false
                 @inbounds for elem_idx in elems_idx_in_box[box_idx]
                     if point ∈ Meshes.element(mesh, elem_idx)
-                        push!(in_mesh_points_idx[id], point_idx)
-                        push!(in_mesh_elements_idx[id], elem_idx)
+                        result_elem_idx[point_idx] = elem_idx
                         found = true
                         break
                     end
                 end
                 if found
-                    # Continue with next point.
                     break
                 end
             end
         end
     end
-    in_mesh_points_idx = reduce(vcat, in_mesh_points_idx)
-    in_mesh_elements_idx = reduce(vcat, in_mesh_elements_idx)
-
-    # _sort_indexes!(in_mesh_points_idx, in_mesh_elements_idx)
+    in_mesh_points_idx = Vector{Int64}()
+    in_mesh_elements_idx = Vector{Int64}()
+    for point_idx in 1:num_points
+        if result_elem_idx[point_idx] != 0
+            push!(in_mesh_points_idx, point_idx)
+            push!(in_mesh_elements_idx, result_elem_idx[point_idx])
+        end
+    end
     in_mesh_points_idx, in_mesh_elements_idx
 end
 

